cmd/senpai/main.go

```diff
@@ -51,16 +53,23 @@ func main() {
 	lastNetID, lastBuffer := getLastBuffer()
 	app.SwitchToBuffer(lastNetID, lastBuffer)
 
+	// Write last buffer on close
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+	go func() {
+		for sig := range sigCh {
+			switch sig {
+			case syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM:
+				app.Close()
+				writeLastBuffer(app)
```

Comment:
This line can be removed, otherwise senpai writes the last buffer twice: once when the signal is received, and another one before `main` returns.